### PR TITLE
[Intel MKL]Fix mkl_softmax integration error when the input tensor is mkl format

### PR DESCRIPTION
**Pull request by mpjlu**
Bugs of master:
The method to get input_dims is wrong when input is mkl tensor.
The layout_type is wrong when input is mkl tensor.

This PR fix these two bugs

### CODE REVIEW

**Comment by qlzh727** [comment]
Please fix the clang format issue. See details in 
https://source.cloud.google.com/results/invocations/71e83fef-47ab-4c32-adf2-d4dce01aa616/targets/%2F%2Ftensorflow%2Ftools%2Fci_build:gen_ci_clang_format_out/log

**Comment by mpjlu** [comment]
clang format is  fixed. Thanks

**Comment by mpjlu** [comment]
clang format is  fixed. Thanks

File: tensorflow/core/kernels/mkl_softmax_op.cc [hunk 1]
```diff
@@ -61,15 +59,28 @@ class MklSoftmaxOp : public OpKernel {
...
+      }
       memory::format layout_type;
       // In MKL, data format passed to mkl softmax op depends on dimension of
-      // the input tensor. Here "x" data format in MKL is used for 1 dim tensor,
-      // "nc" for 2 dim tensor, "tnc" for 3 dim tensor, "nchw" for 4 dim tensor,
-      // and "ncdhw" for 5 dim tensor. Each of the simbols has the following
-      // meaning: n = batch, c = channels, t = sequence length, h = height, w =
-      // width, d = depth
+      // the input tensor.
+      // Here "x" data format in MKL is used for 1 dim tensor, "nc" for 2 dim
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_softmax_op.cc]: Just curious, why do you insert newlines at the beginning of every sentence? Is this intentional?

File: tensorflow/core/kernels/mkl_softmax_op.cc [hunk 2]
```diff
@@ -61,15 +59,28 @@ class MklSoftmaxOp : public OpKernel {
...
+      // tensor,
+      // "tnc" for 3 dim tensor, "nchw" for 4 dim tensor, and "ncdhw" for 5 dim
+      // tensor.
+      // Each of the simbols has the following meaning:
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_softmax_op.cc]:
```suggestion
      // Each of the symbols has the following meaning:
```

File: tensorflow/core/kernels/mkl_softmax_op.cc [hunk 3]
```diff
@@ -81,13 +92,22 @@ class MklSoftmaxOp : public OpKernel {
           layout_type = memory::format::tnc;
           break;
         case 4:
-          layout_type = memory::format::nchw;
+          if (src_mkl_shape.IsMklTensor()) {
+            layout_type = memory::format::nhwc;
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl_softmax_op.cc]: I thought MKL tensors are in NCHW format? Could you please add comments on why the layout is set this way (in the file)?

Also, why does layout matter in softmax? (NHWC vs NCHW, etc.)

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your PR!

File: tensorflow/core/kernels/mkl_softmax_op.cc (hunk 1 above)
Comment (mpjlu) [reply tensorflow/core/kernels/mkl_softmax_op.cc]: This is not intentional, it is generated by clang-format tool. Should this be changed back?

Comment (penpornk) [reply tensorflow/core/kernels/mkl_softmax_op.cc]: I tried applying `clang-format` to the existing file and it didn't reformat this comment block. Could it be that you are using an old version of `clang-format`, or your `clang-format` has a different max-line-length other than 80? 

Could you please change this comment block back (but still fix `simbols` to `symbols`)? Thank you!

File: tensorflow/core/kernels/mkl_softmax_op.cc (hunk 3 above)
Comment (mpjlu) [reply tensorflow/core/kernels/mkl_softmax_op.cc]: Thanks, I will add comments.
When input is MKL tensor, layout_type here is used for output tensor. When input is MKL tensor, output is also MKL tensor, but we need to set its TF layout_type.  TF layout is NHWC.
When input is TF tensor, layout_type here is no special sense. We use axis to define on which dimension to do softmax.

Comment (penpornk) [reply tensorflow/core/kernels/mkl_softmax_op.cc]: Thank you for the explanation! This PR should be good to go after the comments.

**Comment by mpjlu** [comment]
Hi @penpornk @tatianashp , is this PR ready to merge? Thanks.

**Comment by mpjlu** [comment]
Hi @penpornk , the comments are added. Thanks.

**Review by penpornk — APPROVED** [review]
Thank you again for the PR!
